tests/Integration/OtherTest.php

```diff
@@ -277,8 +277,9 @@ public function test_parsely_page_filter(): void {
 		$parsely_options = get_option( \Parsely::OPTIONS_KEY );
 
 		// Create a single post.
-		$post_id = $this->factory->post->create();
-		$post    = get_post( $post_id );
+		$post_array = $this->create_test_post_array();
+		$post_id    = $this->factory->post->create( $post_array );
```

Comment:
What advantage does overriding the [default test post array values](https://core.trac.wordpress.org/browser/trunk/tests/phpunit/includes/factory/class-wp-unittest-factory-for-post.php#L17) here give?

Reply:
The default values create a post with a user that doesn't exist. This can be reverted now that we've changed the `get_author_name` to have a nullable argument.

Reply:
If we're having to provide extra data when creating a test, then it seems like our non-test code should be able to better cope with the data we're passing to `get_author_name()`.

Can we add some more defensive code to the `get_author_names()` (plural) function around line 1804 of `class-parsely.php`, so that we don't pass a `false` to the array_map callback?

This is exactly why we're adding strongly typed code, to catch issues like this. I'm not keen on the current approach of just setting an author, to gloss over this issue.